src/tides_monitoring/rewards_monitor.py

```diff
@@ -0,0 +1,357 @@
+                "created_at": datetime.now(timezone.utc),
+            }
+
+            await db.client.command(earnings_insert, parameters=earnings_params)
+
+            await update_user_balance(
+                db, worker_name, worker_btc_amount, "tides_earnings"
+            )
+
+            logger.debug(
+                f"Processed earnings for {worker_name}: {worker_btc_amount:.8f} BTC "
+                f"({percentage:.2f}% of {total_btc_amount:.8f} BTC)"
+            )
+
+        logger.info(
+            f"Successfully processed TIDES reward {tx_hash} for {len(workers)} workers"
+        )
+
+    except Exception as e:
+        logger.error(f"Failed to process TIDES reward earnings for {tx_hash}: {e}")
+        raise
+
+
+async def update_user_balance(
+    db: StatsDB,
+    worker: str,
+    btc_amount: float,
+    updated_by: str,
+) -> None:
+    """
+    Update user balance by adding to unpaid_amount and total_earned.
+    Uses ClickHouse INSERT with ReplacingMergeTree to handle upserts.
+    """
+    try:
+        current_balance_query = """
+        SELECT unpaid_amount, paid_amount, total_earned
+        FROM user_rewards
+        WHERE worker = %(worker)s
+        ORDER BY last_updated DESC
+        LIMIT 1
+        """
+
+        result = await db.client.query(
+            current_balance_query, parameters={"worker": worker}
+        )
+
+        if result.result_rows:
+            current_unpaid = float(result.result_rows[0][0])
+            current_paid = float(result.result_rows[0][1])
+            current_total_earned = float(result.result_rows[0][2])
+        else:
+            current_unpaid = 0.0
+            current_paid = 0.0
+            current_total_earned = 0.0
+
+        new_unpaid = current_unpaid + btc_amount
+        new_total_earned = current_total_earned + btc_amount
+
+        balance_insert = """
+        INSERT INTO user_rewards (
+            worker, unpaid_amount, paid_amount, total_earned, last_updated, updated_by
+        ) VALUES (
+            %(worker)s, %(unpaid_amount)s, %(paid_amount)s, %(total_earned)s, %(last_updated)s, %(updated_by)s
+        )
+        """
+
+        balance_params = {
+            "worker": worker,
+            "unpaid_amount": new_unpaid,
+            "paid_amount": current_paid,
+            "total_earned": new_total_earned,
+            "last_updated": datetime.now(timezone.utc),
+            "updated_by": updated_by,
+        }
+
+        await db.client.command(balance_insert, parameters=balance_params)
+
+        logger.debug(
+            f"Updated balance for {worker}: +{btc_amount:.8f} BTC (unpaid: {new_unpaid:.8f})"
+        )
+
+    except Exception as e:
+        logger.error(f"Failed to update balance for {worker}: {e}")
+        raise
+
+
+async def load_existing_tx_hashes(db: StatsDB) -> set:
+    """Load all existing tx_hashes from tides_rewards table for duplicate checking."""
+    try:
+        existing_query = "SELECT tx_hash FROM tides_rewards"
```

Comment:
i'm unsure if the tx_hash *will* always be unique. it *should* because of all the OP_RETURN codes. But it's possible that the coinbase outputs are the same may mean the tx_hash is. 

Maybe make a note